### PR TITLE
Feature/add check status

### PR DESCRIPTION
**Pull request by rueyjye**
Add new feature for the checking  status in the detail report.

1. Add new column in test_case_results
2. Separate the update table in show view into partial
3. Add update action in controller
4. Add ajax for update the view asynchronously

### CODE REVIEW

File: app/assets/javascripts/application.js [hunk 1]
```diff
@@ -11,7 +11,7 @@
 // about supported directives.
 //
 //= require jquery
-//= require jquery_ujs
+//= require rails-ujs
```
Comment (rueyjye) [inline app/assets/javascripts/application.js:14]: Use rails-ujs when rails version > 5.1
https://railsguides.jp/working_with_javascript_in_rails.html

File: app/assets/javascripts/pages_main.js [hunk 2]
```diff
@@ -1,27 +1,27 @@
-const labels = ['Passed', 'Failed'];
```
Comment (rueyjye) [inline app/assets/javascripts/pages_main.js:1]: [Bug fix]
Remove the duplicate declaration.

File: app/assets/javascripts/pages_main.js [hunk 3]
```diff
@@ -1,27 +1,27 @@
...
-            'rgba(75, 192, 192, 1)',
-            'rgba(255,99,132,1)',
-          ],
-          borderWidth: 1,
-        }],
-      },
-    });
-  }
-});
+if (gon.data_for_top != void 0) {
```
Comment (rueyjye) [inline app/assets/javascripts/pages_main.js:1]: [Bug fix]
Avoid the bug when gon.data_for_top is undefined.

File: app/assets/javascripts/test_reports_show.js [hunk 4]
```diff
@@ -1,10 +1,7 @@
-const labels = ['Passed', 'Failed'];
```
Comment (rueyjye) [inline app/assets/javascripts/test_reports_show.js:1]: [Bug fix]
Remove the duplicate declaration.

File: app/assets/javascripts/test_reports_show.js [hunk 5]
```diff
@@ -1,10 +1,7 @@
...
-
-const detailChart = 'test-report-detail-chart';
```
Comment (rueyjye) [inline app/assets/javascripts/test_reports_show.js:3]: [Bug fix]
Remove the duplicate declaration.

File: app/assets/javascripts/test_reports_show.js [hunk 6]
```diff
@@ -22,3 +19,19 @@ new Chart(detailChart, {
     }],
   },
 });
+
+var path = window.location.pathname;
+var regex = RegExp('^/test_reports\/[0-9]');
+if (regex.test(path)){
```
Comment (rueyjye) [inline app/assets/javascripts/test_reports_show.js:25]: Let it run only on detail page (url:  fqdn/rest_resport/:id)

File: app/views/test_reports/show.js.erb [hunk 7]
```diff
@@ -0,0 +1,15 @@
+$('.detail_table').html("<%= j(render "detail_table", data_for_test_reports: @data_for_test_reports, select_option: @select_option, job: @job , round: @round) %>");
+
+var element = document.getElementsByName("check_status");
+Object.keys(element).forEach((select) => {
```
Comment (rueyjye) [inline app/views/test_reports/show.js.erb:4]: Change class name asynchronously.
And the element will change the background color by css.

File: app/assets/javascripts/pages_main.js (hunk 3 above)
Comment (rueyjye) [reply app/assets/javascripts/pages_main.js]: remove this

File: app/assets/javascripts/test_reports_show.js (hunk 6 above)
Comment (rueyjye) [reply app/assets/javascripts/test_reports_show.js]: Fix by checking controller and action

File: app/assets/javascripts/pages_main.js [hunk 8]
```diff
@@ -1,27 +1,27 @@
...
-            'rgba(255, 99, 132, 0.5)',
-          ],
-          borderColor: [
-            'rgba(75, 192, 192, 1)',
-            'rgba(255,99,132,1)',
-          ],
-          borderWidth: 1,
-        }],
-      },
+if (gon.controller_name == "pages" && gon.action_name == "main") {
```
Comment (rueyjye) [inline app/assets/javascripts/pages_main.js:1]: Specify the execute page by checking controller and action

File: app/controllers/pages_controller.rb [hunk 9]
```diff
@@ -10,5 +10,7 @@ def main
       end
     end
     gon.data_for_top = @data_for_top
+    gon.controller_name = controller_name
```
Comment (rueyjye) [inline app/controllers/pages_controller.rb:13]: Send the message to javascript for specify the execute page.

File: app/models/test_case_result.rb [hunk 10]
```diff
@@ -35,10 +35,11 @@ class TestCaseResult < ApplicationRecord
   scope :get_total_passed_counts, ->(job_id) { where(job_id: job_id).where(is_error: 0).count }
```
Comment (hikimocr) [inline app/models/test_case_result.rb:35]: Please update upper area.

Comment (rueyjye) [reply app/models/test_case_result.rb:38]: Update by annotate.

**Pull request merged**